src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +42,1136 @@ const ActorInit En_Niw_InitVars = {
     (ActorFunc)EnNiw_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/EnNiw_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/EnNiw_Destroy.s")
+static f32 D_80AB8604[] = {
+    5000.0f,
+    -5000.0f,
+};
+
+static f32 D_80AB860C[] = {
+    5000.0f,
+    3000.0f,
+    4000.0f,
+};
+
+static Vec3f sKakarikoPosList[] = {
+    { -1697.0f, 80.0f, 870.0f }, { 57.0f, 320.0f, -673.0f }, { 796.0f, 80.0f, 1639.0f }, { 1417.0f, 465.0f, 169.0f },
+    { -60.0f, 0.0f, -46.0f },    { -247.0f, 80.0f, 854.0f }, { 1079.0f, 80.0f, -47.0f },
+};
+
+static s16 sKakarikoFlagList[] = {
+    0x0200, 0x0400, 0x0800, 0x1000, 0x2000, 0x4000, 0x8000,
+};
+
+static u8 sSpawnedVer1 = 0;
+
+static u8 sSpawnedVer2 = 0;
+
+static ColliderCylinderInit sCylinderInit1 = {
+    { COLTYPE_UNK5, 0x00, 0x09, 0x01, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 15, 25, 4, { 0, 0, 0 } },
+};
+
+static ColliderCylinderInit sCylinderInit2 = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 15, 25, 4, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 6, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 0, ICHAIN_STOP),
+};
+
+extern SkeletonHeader D_06002530;
+extern AnimationHeader D_060000E8;
+extern Gfx D_060023B0[];
+extern Gfx D_06002428[];
+
+void EnNiw_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiw* this = THIS;
+    s32 pad[3];
+    Vec3f* posListEntry;
+    s32 i;
+
+    if (this->actor.params < 0) {
+        this->actor.params = 0;
+    }
+    if (this->actor.params == 0xB) {
+        if (sSpawnedVer1) {
+            Actor_Kill(&this->actor);
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ もういてる原 Ver.1 ☆☆☆☆☆ \n" VT_RST);
+            return;
+        }
+        sSpawnedVer1 = true;
+        this->actor.room = -1;
+    }
+    if (this->actor.params == 0xC) {
+        if (sSpawnedVer2) {
+            Actor_Kill(&this->actor);
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ もういてる原 Ver.2 ☆☆☆☆☆ \n" VT_RST);
+            return;
+        }
+        sSpawnedVer2 = true;
+        this->actor.room = -1;
+    }
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.flags |= 1;
+    ActorShape_Init(&this->actor.shape, 0.0f, &ActorShadow_DrawFunc_Circle, 25.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06002530, &D_060000E8, this->limbDrawTable,
+                     this->transitionDrawTable, 16);
+
+    if (globalCtx->sceneNum == SCENE_SPOT01) {
+        for (i = 0, posListEntry = sKakarikoPosList; i < ARRAY_COUNT(sKakarikoPosList); i++, posListEntry++) {
+            if (fabsf(this->actor.posRot.pos.x - posListEntry->x) < 40.0f &&
+                fabsf(this->actor.posRot.pos.z - posListEntry->z) < 40.0f) {
```

Comment:
Is it not possible to just do sKakarikoPosList[i] instead of using the posListEntry temp?